integration/tests/access/cohort4/websocket_subscriptions_test.go

```diff
@@ -0,0 +1,1228 @@
+					"arguments":           convertToArguments(tx.Arguments),
+					"reference_block_id":  tx.ReferenceBlockID.String(),
+					"gas_limit":           strconv.FormatUint(tx.GasLimit, 10),
+					"payer":               tx.Payer.String(),
+					"proposal_key":        convertToProposalKey(tx.ProposalKey),
+					"authorizers":         convertToAuthorizers(tx.Authorizers),
+					"payload_signatures":  convertToSig(tx.PayloadSignatures),
+					"envelope_signatures": convertToSig(tx.EnvelopeSignatures),
+				}
+			},
+			listenSubscriptionResponseDuration: 10 * time.Second,
+			testUnsubscribe:                    false,
+		},
+	}
+
+	for _, tt := range tests {
+		// This test cases handles the lifecycle of a websocket connection for a specific subscription,
+		// including sending a subscription and unsubscription requests, listening for incoming responses, and validating
+		// them using a provided validation function.
+		s.Run(tt.name, func() {
+			// Step 1: Establish a WebSocket connection
+			wsClient, err := common.GetWSClient(s.ctx, getWebsocketsUrl(s.restAccessAddress))
+			s.Require().NoError(err)
+			defer func() { s.Require().NoError(wsClient.Close()) }()
+
+			// Step 2: Create and send the subscription request
+			subscriptionRequest := s.subscribeMessageRequest(
+				"dummy_id",
+				tt.topic,
+				tt.prepareArguments(),
+			)
+			s.Require().NoError(wsClient.WriteJSON(subscriptionRequest))
+
+			// Step 3: Listen for WebSocket responses for the specified duration
+			responses, baseMessageResponses, _ := s.listenWebSocketResponses(
+				wsClient,
+				tt.listenSubscriptionResponseDuration,
+				subscriptionRequest.SubscriptionID,
+			)
+
+			// Step 4: Validate the subscription response
+			s.Require().Equal(1, len(baseMessageResponses), "expected one subscription response")
+			s.validateBaseMessageResponse(baseMessageResponses[0])
+
+			// Step 5: Use the provided validation function to check received responses
+			s.validate(
+				subscriptionRequest.SubscriptionID,
+				subscriptionRequest.Topic,
+				responses,
+			)
+
+			// Step 6: Optionally unsubscribe from the topic
+			if tt.testUnsubscribe {
+				// Create an unsubscription request
+				unsubscriptionRequest := s.unsubscribeMessageRequest(subscriptionRequest.SubscriptionID)
+
+				// Send the unsubscription request to the WebSocket server
+				s.Require().NoError(wsClient.WriteJSON(unsubscriptionRequest))
+
+				// Step 6.1: Read and validate the unsubscription response
+				var response models.BaseMessageResponse
+				err := wsClient.ReadJSON(&response)
+				s.Require().NoError(err, "failed to read unsubscription response")
+				s.validateBaseMessageResponse(response)
+			}
+		})
+	}
+}
+
+// testSubscriptionMultiplexing verifies that when subscribing to multiple channels simultaneously,
+// all expected messages are received correctly, ensuring subscription multiplexing works as expected.
+func (s *WebsocketSubscriptionSuite) testSubscriptionMultiplexing() {
```

Comment:
thanks for updating. this is still not quite capturing what I'm after. I opened this PR against this branch to illustrate
https://github.com/The-K-R-O-K/flow-go/pull/3/files

also, when I ran the integration tests locally, the happy case test always failed on "Send and subscribe to transaction statuses". I didn't have time to debug why, but it's possible my machine is running a little faster/slower.